### PR TITLE
Disc refactor

### PR DESCRIPTION
**Pull request by awlego**
Since the disc code was written, there have been a number of improvements to take the suggestions and rendering management out of the CombatLogParser module. I have refactored the disc code to use these new suggestions and rendering management.

### CODE REVIEW

**Comment by rp4rk** [comment]
This is quite large, so bear with me!

File: src/Parser/Priest/Discipline/CHANGELOG.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 export default `
+27-10-2017 - Refactored disc module. (Gao)
```
Comment (awlego) [inline src/Parser/Priest/Discipline/CHANGELOG.js]: Not sure if internal changes that don't have external changes should be in this changelog.

**Comment by awlego** [comment]
Sorry it's large. I tried to think of a reasonable way to break this up... but partial refactor PRs didn't seem great either.

**Comment by rp4rk** [comment]
I've read through this entirely and I'm struggling to find anything worth blocking it, really really impressive work. Thank you!

**Pull request merged**